### PR TITLE
Build jsoncpp from source using jsoncpp.cmake script

### LINKED ISSUE #789: Require jsoncpp 0.9.4 as a minimum
(Follow up from #760)

0.9.4 is the version which has complete UTF8 support and was released in 2015. Earlier versions could output invalid JSON during compilation.

The latest version is 1.7.4, released a month ago.

### PR DESCRIPTION
**Pull request by chfast**
Fixes #789.

### CODE REVIEW

**Comment by axic** [comment]
@chfast the Emscripten build script needs to be updated

**Comment by chriseth** [comment]
Will take a look at emscripten.

**Comment by chfast** [comment]
I have a fix, just waiting for Windows builds to check if everything is ok there.

File: scripts/release_ppa.sh [hunk 1]
```diff
@@ -76,21 +76,6 @@ cp /tmp/${packagename}_${debversion}.orig.tar.gz ../
 
 # Create debian package information
 
-case $distribution in
```
Comment (chriseth) [inline scripts/release_ppa.sh:79]: Hm, haven't thought of ubuntu / debian. Let's hope their build server allows network access.

**Comment by axic** [comment]
It is still failing in Emscripten:

```
Linking CXX static library libjsoncpp.a

Error running link command: No such file or directory

src/lib_json/CMakeFiles/jsoncpp_lib_static.dir/build.make:135: recipe for target 'src/lib_json/libjsoncpp.a' failed

make[5]: *** [src/lib_json/libjsoncpp.a] Error 2
```

and

```
make[3]: warning: jobserver unavailable: using -j1.  Add '+' to parent make rule.
```

**Comment by chfast** [comment]
Yes.

Do you know what `emcmake` and `emmake` do?

**Comment by chriseth** [comment]
The fix is probably to prefix the cmake call in https://github.com/ethereum/cpp-dependencies/compare/f2ede70f33633b26a27299ff39995914db2c6923...ff0c567cc0fd57049b5c29c3b3428097e6ca4644#diff-2f4269495aade38f3a8589ebe150f052R13 with `emcmake`. This does the necessary magic so that the correct compiler and linker is called (and perhaps also other modifications are made).

**Comment by axic** [comment]
@chfast have you got this working?

**Comment by chriseth** [comment]
I tried getting this to work for emscripten at all, but failed. Idea by @chfast: Set `CONFIGURE_COMMAND` to `emcmake cmake` for emscripten and to `cmake` for non-emscripten. Perhaps we also have to remove the install step for emscripten.

**Comment by chriseth** [comment]
Just setting CONFIGURE_COMMAND did no help, now I'm getting

```
CMake Error: The source directory "/src/build/deps/jsoncpp/src/jsoncpp-project-build/Release" does not exist.
Specify --help for usage, or press the help button on the CMake GUI.
ERROR:root:Configure step failed with non-zero return code 1! Command line: ['cmake', '-DCMAKE_CROSSCOMPILING_EMULATOR="nodejs"', '--build', '/src/build/deps/jsoncpp/src/jsoncpp-project-build', '--config', 'Release', '-DCMAKE_TOOLCHAIN_FILE=/emsdk_portable/emscripten/tag-1.35.4/cmake/Modules/Platform/Emscripten.cmake'] at /src/build/deps/jsoncpp/src/jsoncpp-project-build
```

now. Feel free to take this...

**Comment by axic** [comment]
It works! 🎉 

@chfast can this be merged?

**Comment by chfast** [comment]
Fixed! :)
One more build (I accidentally pushed debug messages).

**Review by axic — APPROVED** [review]
It looks good (and builds all targets), but I'm not fully versed in the build system.

**Comment by chfast** [comment]
Are you able to test the Emscripten build before merging?

**Comment by axic** [comment]
Do we need to mention this in the changelog though?

**Comment by chfast** [comment]
Bonus: I removed CryptoPP.

**Pull request merged**